dapps/evancore.vue.libs/src/dispatcher.ts

```diff
@@ -0,0 +1,245 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+
+  You can be released from the requirements of the GNU Affero General Public
+  License by purchasing a commercial license.
+  Buying such a license is mandatory as soon as you use this software or parts
+  of it on other blockchains than evan.network.
+
+  For more information, please contact evan GmbH at this address:
+  https://evan.network/license/
+*/
+
+// vue imports
+import Component, { mixins } from 'vue-class-component';
+import Vue from 'vue';
+import VueRouter from 'vue-router';
+import Vuex from 'vuex';
+import any from 'vuex-i18n';
+import { Prop } from 'vue-property-decorator';
+
+// evan.network imports
+import * as bcc from '@evan.network/api-blockchain-core';
+import * as dappBrowser from '@evan.network/ui-dapp-browser';
+import { EvanQueue, Dispatcher, DispatcherInstance } from '@evan.network/ui';
+import { getDomainName } from './utils';
+
+/**
+ * evan.network vue handler for dispatchers. Adds dispatcher status information to the vuex store.
+ *
+ * @class      EvanVueDispatcher
+ */
+export default class EvanVueDispatcherHandler {
+  /**
+   * Vue instance to access vuex store, ...
+   */
+  vue: Vue;
+
+  /**
+   * Clear active dispatcher watcher.
+   */
+  dispatcherWatcher: any;
+
+  /**
+   * Currents dapp information
+   */
+  dapp: any;
+
+  /**
+   * Currents dapp name
+   */
+  dappName: string;
+
+  /**
+   * List of loaded dispatchers, so dispatcher updates, that were not registered before within this
+   * dispatcher, will be loaded.
+   */
+  loadedDispatchers: Array<string> = [ ];
+
+  /**
+   * Current bcc runtime in user context
+   */
+  runtime: bcc.Runtime;
+
+  /**
+   * All dispatcher instances that are currently in error state, mapped from it's full ens address to a bool flag.
+   */
+  error: any = { };
```

Comment:
i think it would be very helpful to have an DispatcherErrorInterface here. Also for the other types if possible.

Reply:
currently it's just a ens mapping to an error text, so only key value